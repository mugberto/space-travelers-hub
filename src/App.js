import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
import Header from './components/Header';
import MyProfile from './components/myProfile/MyProfile';
import logo from './assets/planet.svg';
import Missions from './components/mission/Missions';
import Rockets from './components/rocket/Rockets';

const routes = [
  {
    path: '/',
    name: 'ROCKETS',
    component: <Rockets />,
  },
  {
    path: '/missions',
    name: 'MISSIONS',
    component: <Missions />,
  },
  {
    path: '/my-profile',
    name: 'My-PROFILE',
    component: <MyProfile />,
  },
];

const App = () => (
  <Router>

    <Header logo={logo} routes={routes} />
    <Switch>
      {routes.map(({ path, component }) => (
        <Route path={path} exact key={path}>{component}</Route>
      ))}
    </Switch>
  </Router>
);

export default App;
