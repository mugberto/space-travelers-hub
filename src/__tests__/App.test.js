import React from 'react';
import { render, fireEvent, screen } from '../test-utils';
import App from '../App';

describe('App', () => {
  it('renders page correctly', () => {
    const tree = render(<App />);
    expect(tree).toMatchSnapshot();
  });

  it('it loads "Rockets" on first render', async () => {
    render(<App />);
    expect(await screen.findByText('Falcon 1')).toBeInTheDocument();
  });

  it('navigates to Missions and displays missions', async () => {
    render(<App />);
    fireEvent.click(screen.getByRole('link', { name: 'Missions' }));
    expect(await screen.findByText('Thaicom')).toBeInTheDocument();
  });

  it('navigates to "Profile" and displays user subscriptions', async () => {
    render(<App />);
    fireEvent.click(screen.getByRole('link', { name: 'My Profile' }));
    expect(await screen.findByText('NO MISSIONS JOINED YET')).toBeInTheDocument();
    expect(await screen.findByText('NO ROCKET BOOKED YET')).toBeInTheDocument();
  });
});
