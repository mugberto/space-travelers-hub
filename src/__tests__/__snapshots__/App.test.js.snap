// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders page correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="brand-logo"
        >
          <img
            alt="logo"
            height="40"
            src="planet.svg"
            width="40"
          />
          <span>
            Space Travelers' Hub
          </span>
        </div>
        <nav>
          <ul
            class="nav-links"
          >
            <li
              class="nav-link"
            >
              <a
                aria-current="page"
                class="nav-link active"
                href="/"
              >
                Rockets
              </a>
            </li>
            <li
              class="nav-link"
            >
              <a
                class="nav-link"
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li
              class="nav-link"
            >
              <a
                class="nav-link"
                href="/my-profile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </nav>
      </header>
      <div
        class="MuiBox-root css-0"
        style="margin: 0px 28px;"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="brand-logo"
      >
        <img
          alt="logo"
          height="40"
          src="planet.svg"
          width="40"
        />
        <span>
          Space Travelers' Hub
        </span>
      </div>
      <nav>
        <ul
          class="nav-links"
        >
          <li
            class="nav-link"
          >
            <a
              aria-current="page"
              class="nav-link active"
              href="/"
            >
              Rockets
            </a>
          </li>
          <li
            class="nav-link"
          >
            <a
              class="nav-link"
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li
            class="nav-link"
          >
            <a
              class="nav-link"
              href="/my-profile"
            >
              My Profile
            </a>
          </li>
        </ul>
      </nav>
    </header>
    <div
      class="MuiBox-root css-0"
      style="margin: 0px 28px;"
    >
      <ul
        class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
